src/test/kotlin/io/github/rybalkinsd/kohttp/interceptors/SigningInterceptorTest.kt

```diff
@@ -0,0 +1,43 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import io.github.rybalkinsd.kohttp.client.defaultHttpClient
+import io.github.rybalkinsd.kohttp.client.fork
+import io.github.rybalkinsd.kohttp.dsl.httpGet
+import io.github.rybalkinsd.kohttp.util.asJson
+import org.junit.Test
+import java.security.MessageDigest
+import java.util.*
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class SigningInterceptorTest {
+    @Test
+    fun `signs request with MD5 algorithm`() {
+        val urlEncoder = Base64.getUrlEncoder()
+        val algorithm = MessageDigest.getInstance("md5")
+
+        val client = defaultHttpClient.fork {
+            interceptors = listOf(SigningInterceptor("key") {
+                val query = (query() ?: "").toByteArray()
```

Comment:
This peace of code is most related to user experience. It should be as easy to interact with as possible.
 
Possible suggestions here:
- We can introduce extension function for `MessageDigest` to encode `String?` in `MessageDigestExt` file
- It's also possible to provide a util function like `md5(data: String?)`

From my experience it's also may be useful to provide ext to operate with `Map<String, List<String>` (map of param names to their values.

Reply:
I'll check with android buddies what they are using tomorrow :)

Reply:
> Do we provide this on the Interceptor?
three options here:
1. change `signer` `HttpUrl.() -> String` to `Map<String, List<String>> -> String`
2. support both
3. support both + `String -> String` ("foo=bar&random=213" -> hash(...))

Reply:
Let's have ext only for `hmac` and `md5` now.

Reply:
> > Do we provide this on the Interceptor?
> > three options here:
> 
> 1. change `signer` `HttpUrl.() -> String` to `Map<String, List<String>> -> String`
> 2. support both
> 3. support both + `String -> String` ("foo=bar&random=213" -> hash(...))

> Let's have ext only for `hmac` and `md5` now.

I haven't had a chance to work on these yet, I will have a think about design and raise a PR for it.